core/services/feeds/orm.go

```diff
@@ -103,3 +106,79 @@ func (o *orm) CountManagers() (int64, error) {
 
 	return count, nil
 }
+
+// CreateJobProposal creates a job proposal.
+func (o *orm) CreateJobProposal(ctx context.Context, jp *JobProposal) (uint, error) {
+	var id uint
+	now := time.Now()
+
+	stmt := `
+		INSERT INTO job_proposals (spec, status, job_id, feeds_manager_id, created_at, updated_at)
+		VALUES (?, ?, ?, ?, ?, ?)
+		RETURNING id;
+	`
+
+	row := o.db.Raw(stmt, jp.Spec, jp.Status, jp.JobID, jp.FeedsManagerID, now, now).Row()
+	if row.Err() != nil {
+		return id, row.Err()
+	}
+
+	err := row.Scan(&id)
+	if err != nil {
+		return id, err
+	}
+
+	return id, err
+}
+
+// ListJobProposals lists all job proposals
+func (o *orm) ListJobProposals(ctx context.Context) ([]JobProposal, error) {
+	jps := []JobProposal{}
+	stmt := `
+		SELECT id, spec, status, job_id, feeds_manager_id, created_at, updated_at
+		FROM job_proposals;
+	`
+
+	err := o.db.Raw(stmt).Scan(&jps).Error
+	if err != nil {
+		return jps, err
+	}
+
+	return jps, nil
+}
+
+// GetJobProposal gets a job proposal by id
+func (o *orm) GetJobProposal(ctx context.Context, id uint) (*JobProposal, error) {
+	stmt := `
+		SELECT id, spec, status, job_id, feeds_manager_id, created_at, updated_at
+		FROM job_proposals
+		WHERE id = ?;
+	`
+
+	jp := JobProposal{}
+	result := o.db.Raw(stmt, id).Scan(&jp)
+	if result.RowsAffected == 0 {
+		return nil, sql.ErrNoRows
+	}
+	if result.Error != nil {
+		return nil, result.Error
+	}
+
+	return &jp, nil
+}
+
+// CountJobProposals counts the number of job proposal records.
+func (o *orm) CountJobProposals() (int64, error) {
+	var count int64
+	stmt := `
+		SELECT COUNT(*)
+		FROM job_proposals
```

Comment:
Roughly how big do we expect this table to get?

Reply:
It would most likely be less that the average number of  jobs that a node runs, however I'm only using this for tests to ensure a record was created. I don't see us actually needing to return the count.